nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py

```diff
@@ -0,0 +1,232 @@
+            all_input_mask[i] = all_input_mask[i] + [0] * extra
+
+            if with_label:
+                all_slots[i] = all_slots[i] + [pad_label] * extra
+
+        all_segment_ids.append([0] * max_seq_length)
+
+    logging.info(f'{too_long_count} are longer than {max_seq_length}')
+
+    logging.info("*** Some Examples of Processed Data***")
+    for i in range(min(len(all_input_ids), 5)):
+        logging.info("i: %s" % (i))
+        logging.info("subtokens: %s" % " ".join(list(map(str, all_subtokens[i]))))
+        logging.info("loss_mask: %s" % " ".join(list(map(str, all_loss_mask[i]))))
+        logging.info("input_mask: %s" % " ".join(list(map(str, all_input_mask[i]))))
+        logging.info("subtokens_mask: %s" % " ".join(list(map(str, all_subtokens_mask[i]))))
+        if with_label:
+            logging.info("slots_label: %s" % " ".join(list(map(str, all_slots[i]))))
+
+    return (all_input_ids, all_segment_ids, all_input_mask, all_loss_mask, all_subtokens_mask, all_slots)
+
+
+class IntentSlotClassificationDataset(Dataset):
+    """
+    Creates dataset to use for the task of joint intent
+    and slot classification with pretrained model.
+
+    Converts from raw data to an instance that can be used by
+    NMDataLayer.
+
+    For dataset to use during inference without labels, see
+    IntentSlotDataset.
+
+    Args:
+        input_file: file to sequence + label. the first line is header (sentence [tab] label)
+            each line should be [sentence][tab][label]
+        slot_file: file to slot labels, each line corresponding to slot labels for a sentence in input_file. No header.
+        max_seq_length: max sequence length minus 2 for [CLS] and [SEP]
+        tokenizer: such as NemoBertTokenizer
+        num_samples: number of samples you want to use for the dataset. If -1, use all dataset. Useful for testing.
+        pad_label: pad value use for slot labels. by default, it's the neutral label.
+        ignore_extra_tokens: whether to ignore extra tokens in the loss_mask.
+        ignore_start_end: whether to ignore bos and eos tokens in the loss_mask.
+        do_lower_case: convert query to lower case or not
+    """
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+               """
+        return {
+            'input_ids': NeuralType(('B', 'T'), ChannelType()),
+            'segment_ids': NeuralType(('B', 'T'), ChannelType()),
+            'input_mask': NeuralType(('B', 'T'), MaskType()),
+            'loss_mask': NeuralType(('B', 'T'), MaskType()),
+            'subtokens_mask': NeuralType(('B', 'T'), MaskType()),
+            'intent_labels': NeuralType(('B', 'T'), LabelsType()),
+            'slot_labels': NeuralType(('B', 'T'), LabelsType()),
+        }
+
+    def __init__(
+        self,
+        input_file: str,
+        slot_file: str,
+        max_seq_length: int,
+        tokenizer: Any,
```

Comment:
tokenizer is not Any, but TokenizerSpec, please see https://github.com/NVIDIA/NeMo/blob/candidate/nemo/collections/nlp/data/punctuation_capitalization_dataset.py#L35

Reply:
Ok. In Vahid code (from where I copied it, is Any).

Reply:
And Evelina in your TokenClassificationDataSet there is no types at all in Init

Reply:
you're right, and I'm going to fix that, thanks for pointing this out

Reply:
Corrected